integration-tests/src/test/java/com/datastax/oss/driver/api/core/ProtocolVersionMixedClusterIT.java

```diff
@@ -0,0 +1,186 @@
+                "SELECT cluster_name FROM system.local",
+                "SELECT * FROM system.local",
+                "SELECT * FROM system.peers");
+      }
+    }
+  }
+
+  @Test
+  public void should_keep_current_if_supported_by_all_peers() {
+    try (BoundCluster simulacron = mixedVersions("3.0.0", "2.2.0", "3.11");
+        BoundNode contactPoint = simulacron.node(0);
+        Cluster cluster =
+            Cluster.builder()
+                .addContactPoint(contactPoint.inetSocketAddress())
+                .withConfigLoader(new TestConfigLoader())
+                .build()) {
+
+      InternalDriverContext context = (InternalDriverContext) cluster.getContext();
+      assertThat(context.protocolVersion()).isEqualTo(CoreProtocolVersion.V4);
+      assertThat(simulacron.getLogs().getQueryLogs()).hasSize(3);
+      assertThat(contactPoint.getLogs().getQueryLogs().stream().map(QueryLog::getQuery))
+          .containsExactly(
+              // Initial connection with protocol v4
+              "SELECT cluster_name FROM system.local",
+              "SELECT * FROM system.local",
+              "SELECT * FROM system.peers");
+    }
+  }
+
+  @Test
+  public void should_fail_if_peer_does_not_support_v3() {
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage(
+        "reports Cassandra version 2.0.9, but the driver only supports 2.1.0 and above");
+
+    try (BoundCluster simulacron = mixedVersions("3.0.0", "2.0.9", "3.11");
+        BoundNode contactPoint = simulacron.node(0);
+        Cluster ignored =
+            Cluster.builder()
+                .addContactPoint(contactPoint.inetSocketAddress())
+                .withConfigLoader(new TestConfigLoader())
+                .build()) {
+      fail("Cluster init should have failed");
+    }
+  }
+
+  @Test
+  public void should_not_downgrade_and_force_down_old_nodes_if_version_forced() {
+    try (BoundCluster simulacron = mixedVersions("3.0.0", "2.2.0", "2.0.0");
+        BoundNode contactPoint = simulacron.node(0);
+        Cluster cluster =
+            Cluster.builder()
+                .addContactPoint(contactPoint.inetSocketAddress())
+                .withConfigLoader(new TestConfigLoader("protocol.version = V4"))
+                .build()) {
+      assertThat(cluster.getContext().protocolVersion()).isEqualTo(CoreProtocolVersion.V4);
+
+      assertThat(simulacron.getLogs().getQueryLogs()).hasSize(3);
+      assertThat(contactPoint.getLogs().getQueryLogs().stream().map(QueryLog::getQuery))
+          .containsExactly(
+              // Initial connection with protocol v4
+              "SELECT cluster_name FROM system.local",
+              "SELECT * FROM system.local",
+              "SELECT * FROM system.peers");
+
+      // Note: the 2.0.0 would be forced down if we try to open a connection to it. We can't check
+      // that here because Simulacron can't prime STARTUP requests.
+    }
+  }
+
+  private BoundCluster mixedVersions(String... versions) {
```

Comment:
Nice!  The simulacron cluster builders are kinda ugly to work with, nice to simplify this with a method.